Source/Core/Core/AchievementManager.cpp

```diff
@@ -669,17 +672,20 @@ void AchievementManager::DoFrame()
   {
     m_framecount++;
   }
-  Core::RunAsCPUThread([&] {
-    rc_runtime_do_frame(
-        &m_runtime,
-        [](const rc_runtime_event_t* runtime_event) {
-          GetInstance().AchievementEventHandler(runtime_event);
-        },
-        [](unsigned address, unsigned num_bytes, void* ud) {
-          return static_cast<AchievementManager*>(ud)->MemoryPeeker(address, num_bytes, ud);
-        },
-        this, nullptr);
-  });
+  {
+    std::lock_guard lg{m_lock};
+    Core::RunAsCPUThread([&] {
```

Comment:
This smells dangerous. `DoFrame()` should always run on the CPU thread anyway though, yeah? At least it does right now and I see no obvious reason it should run elsewhere. So maybe just remove the `RunAsCPUThread()` and add an `ASSERT(IsCPUThread());` at the start of the function.